lib/core/src/swapper/mod.rs

```diff
@@ -147,15 +148,38 @@ pub trait Swapper: Send + Sync {
 
 pub struct BoltzSwapper {
     client: BoltzApiClientV2,
+    referral_id: Option<String>,
     config: Config,
     liquid_electrum_config: ElectrumConfig,
     bitcoin_electrum_config: ElectrumConfig,
 }
 
 impl BoltzSwapper {
-    pub fn new(config: Config) -> BoltzSwapper {
+    pub fn new(config: Config, swapper_proxy_url: Option<String>) -> BoltzSwapper {
+        let (boltz_api_base_url, referral_id) = match &config.network {
+            LiquidNetwork::Testnet => (None, None),
+            LiquidNetwork::Mainnet => match &swapper_proxy_url {
+                Some(swapper_proxy_url) => Url::parse(swapper_proxy_url)
+                    .map(|url| match url.query() {
+                        None => (None, None),
+                        Some(query) => {
+                            let api_base_url =
+                                url.domain().map(|domain| format!("https://{domain}/v2"));
+                            let parts: Vec<String> = query.split('=').map(Into::into).collect();
+                            let referral_id = parts.get(1).cloned();
+                            (api_base_url, referral_id)
+                        }
+                    })
+                    .unwrap_or_default(),
+                None => (None, None),
+            },
+        };
+
+        let boltz_url = boltz_api_base_url.unwrap_or(config.boltz_url.clone());
```

Comment:
I think we should remove the boltz_url from the config as we take it from the breez server and the user won't understand why it is not in use. It is actually sort of fallback so I suggest to have it as constant inside the swapper and not expose it in the configuration.

Reply:
Done: d7634421ca7b7fdcf4d2437331b1e15f3de9bab8